app/controllers/events_controller.rb

```diff
@@ -0,0 +1,81 @@
+class EventsController < ApplicationController
+  before_action :set_event, only: %i[show edit update destroy]
+  before_action :authenticate_user!, except: %i[index show]
+
+  # GET /events
+  # GET /events.json
+  def index
+    @events = Event.all
```

Comment:
instead of having one variable events use upcoming_events and past_events variables, like so:
- @upcaming_events = Event.upcoming_events
- @past_events = Event.previous_events